sdcm/sct_config.py

```diff
@@ -1222,6 +1222,10 @@ class SCTConfiguration(dict):
         dict(name="availability_zone", env="SCT_AVAILABILITY_ZONE",
              type=str,
              help="Availability zone to use. Same for multi-region scenario."),
+        dict(name="auto_availability_zone", env="SCT_AWS_AUTO_AVAILABILITY_ZONE",
+             type=boolean,
+             help="Not limiting availability zones, in order for the cloud provider to automatically select"
```

Comment:
Since it is backend-specific the description must mention it.
But, I guess @soyacz will have concerns because of the provision logic unification process violation.

Reply:
right, that's the cocern i had as well, that although only implement currently for AWS, it can be similarly added in the future to other providers.

Reply:
@soyacz , could you please review this PR?

Reply:
can we just not set availability_zone and consider automatic if not set instead of adding new variable?

Reply:
This whole automatic is not the requirement and not part of the task.

Reply:
@roydahan , @fgelcer , there can be 2 different ways for a solution:
1) Catch 'no resources' and retry another zone. (As stated in AWS error message)
2) Not sending AZ related requirements to AWS (as recommended in AWS error message as well). This is what asked in issue 5519 and by Lukasz above.
So we have to decide one of the above.

Reply:
The solution is stated in the task description.
We considered other solutions and don't think they're a good fit.

Reply:
fixed